src/server/database/Database/DatabaseWorkerPool.cpp

```diff
@@ -1,33 +1,65 @@
 /*
- * Copyright (C) 2016+     AzerothCore <www.azerothcore.org>
- * Copyright (C) 2008-2016 TrinityCore <http://www.trinitycore.org/>
- * Copyright (C) 2005-2009 MaNGOS <http://getmangos.com/>
+ * Copyright (C) 2016+ AzerothCore <www.azerothcore.org>, released under GNU GPL v2 license, you may redistribute it and/or modify it under version 2 of the License, or (at your option), any later version.
+ * Copyright (C) 2021+ WarheadCore <https://github.com/WarheadCore>
  */
 
 #include "DatabaseWorkerPool.h"
-#include "DatabaseEnv.h"
-
+#include "AdhocStatement.h"
+#include "Common.h"
+#include "Errors.h"
+#include "Implementation/CharacterDatabase.h"
+#include "Implementation/LoginDatabase.h"
+#include "Implementation/WorldDatabase.h"
+#include "Log.h"
+#include "MySQLPreparedStatement.h"
+#include "MySQLWorkaround.h"
+#include "PreparedStatement.h"
+#include "ProducerConsumerQueue.h"
+#include "QueryCallback.h"
+#include "QueryHolder.h"
+#include "QueryResult.h"
+#include "SQLOperation.h"
+#include "Transaction.h"
 #include <mysqld_error.h>
 
+#ifdef ACORE_DEBUG
+#include <boost/stacktrace.hpp>
+#include <sstream>
+#endif
+
 #define MIN_MYSQL_SERVER_VERSION 50700u
 #define MIN_MYSQL_CLIENT_VERSION 50700u
 
-template <class T> DatabaseWorkerPool<T>::DatabaseWorkerPool() :
-    _mqueue(new ACE_Message_Queue<ACE_SYNCH>(2 * 1024 * 1024, 2 * 1024 * 1024)),
-    _queue(new ACE_Activation_Queue(_mqueue)),
-    _async_threads(0),
-    _synch_threads(0)
+class PingOperation : public SQLOperation
 {
-    memset(_connectionCount, 0, sizeof(_connectionCount));
-    _connections.resize(IDX_SIZE);
+    //! Operation for idle delaythreads
+    bool Execute() override
+    {
+        m_conn->Ping();
+        return true;
+    }
+};
 
+template <class T>
+DatabaseWorkerPool<T>::DatabaseWorkerPool()
+    : _queue(new ProducerConsumerQueue<SQLOperation*>()),
+      _async_threads(0), _synch_threads(0)
+{
     WPFatal(mysql_thread_safe(), "Used MySQL library isn't thread-safe.");
     WPFatal(mysql_get_client_version() >= MIN_MYSQL_CLIENT_VERSION, "AzerothCore does not support MySQL versions below 5.7");
+    WPFatal(mysql_get_client_version() == MYSQL_VERSION_ID, "Used MySQL library version (%s id %lu) does not match the version id used to compile AzerothCore (id %u)",
+        mysql_get_client_info(), mysql_get_client_version(), MYSQL_VERSION_ID);
```

Comment:
@FrancescoBorzi @Sedatyf 
Here is what is breaking it for you. And this is intentional as having a lib version != the version of MySQL that was used to compile the core could cause issues.

This is good behavior to prevent issues that could come. It only makes sense.

https://github.com/azerothcore/azerothcore-wotlk/pull/5611#issuecomment-860242128

Reply:
I will say again, it would be if I used MySQL library from 5.7 when I have compiled with MySQL 8.0, it does not make sense and the installation guide also points this out.

Reply:
I compiled myself, got the same error, followed the libmysql.dll part of https://www.azerothcore.org/wiki/windows-core-installation#compiling-the-source and now it works :)

Reply:
makes sense, thanks for the clarification @Kitzunu 

cc @Sedatyf